### PR TITLE
Re-enable ignored simtests

### PR DESCRIPTION
**Pull request by sadhansood**
## Description 

Describe the changes or additions included in this PR.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-benchmark/src/lib.rs [hunk 1]
```diff
@@ -264,6 +264,9 @@ impl ValidatorProxy for LocalValidatorAggregatorProxy {
     }
 
     async fn execute_transaction(&self, tx: Transaction) -> anyhow::Result<ExecutionEffects> {
+        if std::env::var("BENCH_MODE").is_ok() {
```
Comment (mwtian) [inline crates/sui-benchmark/src/lib.rs:267]: So does this require `BENCH_MODE` to be set outside of simtests? Maybe instead we can run `execute_transaction()` only in tests and simtests?

**Review by mwtian — COMMENTED** [review]
Otherwise I think falling back to quorum driver in simtest is a good approach!

File: crates/sui-benchmark/tests/simtest.rs [hunk 2]
```diff
@@ -68,7 +67,6 @@ mod test {
     }
 
     #[sim_test(config = "test_config()")]
-    #[ignore]
```
Comment (mystenmark) [inline crates/sui-benchmark/tests/simtest.rs:71]: the ignores that don't have the long comments should actually be left in place. these tests work but are not stable enough to run in CI.

Comment (mwtian) [reply crates/sui-benchmark/tests/simtest.rs:71]: Is this still unstable after fixing the Narwhal issue?

File: crates/sui-benchmark/src/lib.rs (hunk 1 above)
Comment (sadhansood) [reply crates/sui-benchmark/src/lib.rs:267]: Yes, this change explicitly wants the user to set `BENCH_MODE ` if they want to use `execute_bench_transaction()` which does not handle epoch changes like the quorum driver. So, the default behavior is favoring correctness (without the env variable set) and allows user to switch to more performant code path if they want to.

File: crates/sui-benchmark/tests/simtest.rs (hunk 2 above)
Comment (sadhansood) [reply crates/sui-benchmark/tests/simtest.rs:71]: ok, i will put those back

File: crates/sui-benchmark/src/lib.rs (hunk 1 above)
Comment (sadhansood) [reply crates/sui-benchmark/src/lib.rs:267]: For instance I encountered this locally that stress client started failing after epoch change when i used a small epoch duration

Comment (mwtian) [reply crates/sui-benchmark/src/lib.rs:267]: This is good point. I think this gets into the debate for whether stress client is intended for benchmarking or generating transactions load. If we require users to set `BENCH_MODE`, I think @asonnino should be aware so he can add the flag when running tests. Also we need to add the flag when running stress client in `sui-operations`.

**Review by mwtian — APPROVED** [review]
Rebase should fix lint.

**Pull request merged**